base/sparse/sparsematrix.jl

```diff
@@ -2250,10 +2250,8 @@ setindex!(A::SparseMatrixCSC, x, i::Union{Integer, AbstractVector}, ::Colon) = s
 function setindex!{Tv}(A::SparseMatrixCSC{Tv}, x::Number,
         I::AbstractVector{<:Integer}, J::AbstractVector{<:Integer})
     if isempty(I) || isempty(J); return A; end
-    if !issorted(I); I = sort(I); end
-    if !issorted(J); J = sort(J); end
-    if !allunique(I); I = unique(I); end
-    if !allunique(J); J = unique(J); end
+    if !issorted(I, lt=≤); I = !allunique(I) ? sort!(unique(I)) : sort(I); end
```

Comment:
I think there was an issue on this a while back - probably a good idea to at least add a comment that `by=≤` is testing for strict sorting